README.md

```diff
@@ -12,61 +12,35 @@ Shopify Application Rails engine and generator
 
```

Comment:
I reorganized the readme to move all authentication and session related content into one heading called "Authentication" and then I added new content around user-based token strategy in there.

Reply:
@tanema @tylerball Are you guys ok with this reorganization change here?

Reply:
Looks good! Thanks for cleaning up the TOC